contracts/PetPark.sol

```diff
@@ -2,6 +2,72 @@
 pragma solidity ^0.8.0;
 
 
+
 contract PetPark {
 
+    event Added(uint animalType, uint count);
+    event Borrowed(uint animalType);
+    event Returned(uint animalType);
+
+    address public owner;
+    mapping(uint => uint) shelter;
+    mapping(address=>uint) borrowedAnimals;
+    mapping(address=>mapping(string=>uint)) borrowers;
+
+    // need to make sure that the borrower from the contract always send the same age and gender
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function add(uint animalType, uint count) public {
+        require(msg.sender == owner, "Not an owner");
+        require(animalType>=1 && animalType <= 5,"Invalid animal");
+        shelter[animalType] += count;
+        emit Added(animalType, count);
+    }
+
+    function animalCounts(uint animalType) public view returns (uint){
+        return shelter[animalType];
+    }
+
+    function borrow(uint age, uint gender, uint animalType) public {
+        if (borrowers[msg.sender]["age"] == 0 && borrowers[msg.sender]["gender"] == 0) {
```

Comment:
Since `borrowers[msg.sender]` is being used multiple times it can be stored in a single memory variable. Later in the course you will learn that it optimises gas usage